tests/parametric/test_otel_span_methods.py

```diff
@@ -348,3 +377,93 @@ def test_otel_get_span_context(self, test_agent, test_library):
                     assert context.get("trace_id") == parent.span_context().get("trace_id")
                     assert context.get("span_id") == "{:016x}".format(span.span_id)
                     assert context.get("trace_flags") == "01"
+
+    @missing_feature(context.library == "golang", reason="Not implemented")
+    @missing_feature(context.library == "java", reason="Not implemented")
+    @missing_feature(context.library == "nodejs", reason="Not implemented")
+    @missing_feature(context.library == "dotnet", reason="Not implemented")
+    @missing_feature(context.library == "python", reason="Not implemented")
+    @missing_feature(context.library == "python_http", reason="Not implemented")
+    @pytest.mark.parametrize(
+        "expected_operation_name,span_kind,attributes",
+        [
+            ("http.server.request", SK_SERVER, {"http.request.method": "GET"}),
+            ("http.client.request", SK_CLIENT, {"http.request.method": "GET"}),
+            ("redis.query", SK_CLIENT, {"db.system": "Redis"}),
+            ("kafka.receive", SK_CLIENT, {"messaging.system": "Kafka", "messaging.operation": "Receive"}),
+            ("kafka.receive", SK_SERVER, {"messaging.system": "Kafka", "messaging.operation": "Receive"}),
+            ("kafka.receive", SK_PRODUCER, {"messaging.system": "Kafka", "messaging.operation": "Receive"}),
+            ("kafka.receive", SK_CONSUMER, {"messaging.system": "Kafka", "messaging.operation": "Receive"}),
+            ("aws.s3.request", SK_CLIENT, {"rpc.system": "aws-api", "rpc.service": "S3"}),
+            ("aws.client.request", SK_CLIENT, {"rpc.system": "aws-api"}),
+            ("grpc.client.request", SK_CLIENT, {"rpc.system": "GRPC"}),
+            ("grpc.server.request", SK_SERVER, {"rpc.system": "GRPC"}),
+            ("aws.my-function.invoke", SK_CLIENT, {"faas.invoked_provider": "aws", "faas.invoked_name": "My-Function"}),
+            ("datasource.invoke", SK_SERVER, {"faas.trigger": "Datasource"}),
+            ("graphql.server.request", SK_SERVER, {"graphql.operation.type": "query"}),
+            ("amqp.server.request", SK_SERVER, {"network.protocol.name": "Amqp"}),
+            ("server.request", SK_SERVER, None),
+            ("amqp.client.request", SK_CLIENT, {"network.protocol.name": "Amqp"}),
+            ("client.request", SK_CLIENT, None),
+            ("internal", SK_INTERNAL, None),
+            ("consumer", SK_CONSUMER, None),
+            ("producer", SK_PRODUCER, None),
+            ("otel_unknown", None, None),
+        ],
+    )
+    def test_otel_span_operation_name(
+        self, expected_operation_name: str, span_kind: int, attributes: dict, test_agent, test_library
+    ):
+        run_operation_name_test(
+            resource="otel_span_name",
```

Comment:
nit: if you are always using the same resource name, then just use it in `run_operation_name_test` and don't make it a parameter. i.e. only needs to be a parameter if it's ever different.